source/Nuke.SourceGenerators/AccessInheritanceAsMembersGenerator.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright 2023 Maintainers of NUKE.
+// Distributed under the MIT License.
+// https://github.com/nuke-build/nuke/blob/master/LICENSE
+
+using System;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Nuke.Common.Utilities.Collections;
+
+namespace Nuke.SourceGenerators;
+
+[Generator]
+public class AccessInheritanceAsMembersGenerator : ISourceGenerator
+{
+    public void Initialize(GeneratorInitializationContext context)
+    {
+    }
+
+    public void Execute(GeneratorExecutionContext context)
+    {
+        try
+        {
+            if (GenerateSolutionSource(context.Compilation) is { Length: > 0 } generatedSource)
+            {
+                context.AddSource(nameof(AccessInheritanceAsMembersGenerator), generatedSource);
+            }
+        }
+        catch (Exception exception)
+        {
+            var diagnostic = Diagnostic.Create(
+                "NUKE001",
```

Comment:
Since this is another source generator, the diagnostic id should not be the same.